analysis/spectra_chisq/_model_spectra.py

```diff
@@ -0,0 +1,475 @@
+
+    Note: DR1 neglects extinction from host galaxy, so only MW dust is modeled currently
+    Note: typical Av ~ 0.42 mag via Av = Rv * E(B-V) = 3.1 * 0.134mag ~ 0.42mag
+    where Milky Way ebv = 0.134 (from Table 8 of DR3/Folatelli et al. 2013)
+    and Milky Way Rv = 3.1 from DR1 at the end of pg 9
+    
+    Args:
+        model (Model): sncosmo dust model
+        obj_id (str):
+        redshift (float)
+        amplitude (float):
+        a_v (float): visual (V filter) extinction in units of magnitude
+        r_v (float):
+    
+    Returns:
+        An updated sncosmo model
+    """
+    
+    ebv = a_v / r_v # unit: mags
+    # Get time of peak brightness in JD
+    t0 = get_jd_t0(obj_id=obj_id)
+    # From the example of `model.set()` on https://sncosmo.readthedocs.io/en/v1.6.x/models.html,
+    # it looks like t0 is in JD
+    model.set(z=redshift, t0=t0, mwebv=ebv, amplitude=amplitude)
+    
+    return model
+
+
+def get_model_flux(wave, amplitude, a_v, obj_id, time, model, redshift):
+    """Function to use for fitting amplitude and Av of model.
+    Note: order of args determines which parameters are fit
+    
+    Args:
+        wave (1d array-like of floats): wavelengths in units of Angstrom
+        amplitude (float): fit parameter
+        a_v (float): fit parameter
+        photo_data (Table):
+        redshift (float):
+        obj_id:
+        time (float or array): JD
+        model (sncosmo dust model):
+        
+    Returns:
+        The fluxes (as a 1d array-like) at each wavelength in `wave` computed using `model.` Units: ergs/s/cm^2/Angstrom
+    """
+
+    model = create_model(model=model, obj_id=obj_id, a_v=a_v, redshift=redshift, amplitude=amplitude)
+
+    # https://sncosmo.readthedocs.io/en/v1.1.x/models.html
+    # Get spectrum at a given observer-frame time and wavelengths (spectrum contains OBSERVER frame flux)
+    # Note from docs: can supply a list or array of times and get a 2-d array back,
+    # representing the spectrum at each time
+    mdl_flux = model.flux(time, wave) # Units: ergs/s/cm^2/Angstrom
+    
+    return mdl_flux
+
+
+def plot_spectra_and_residuals(x1, y1, x2, y2):
+    """Make two panel plot. Upper plot: data and model flux vs wavelength. Lower plot: residuals (data - model)
+    
+    Args:
+        x1 (1d array-like of floats): Observer frame wavelength
+        y1 (1d array-like of floats): Observed flux
+        x2 (1d array-like of floats): Observer frame wavelength
+        y2 (1d array-like of floats): Modeled flux
+    """
+    
+    # Set figure size
+    plt.rcParams['figure.figsize'] = [12, 10] # units: inches
+    # Create two stacked figures
+    f, (ax1, ax2) = plt.subplots(2, 1, sharex=True, gridspec_kw={'hspace':0, 'width_ratios':[1]})
+    
+    # Upper plot: flux vs wavelength
+    # Plot wavelength vs obs flux
+    ax1.plot(x1, y1, linestyle=' ', marker='.', label='obs')
```

Comment:
Here and throughout use a single blank line to distinguish separate conceptual steps.

```python
    # Upper plot: flux vs wavelength
    ax1.plot(x1, y1, linestyle=' ', marker='.', label='obs')
    ax1.plot(x2, y2, linestyle=' ', marker='.', label='fit')
    ax1.set_ylabel('flux [ergs/(s cm^2 Angstrom)]', fontsize=14)
    ax1.legend(fontsize=14)
    
    # Lower plot: residuals (data - model) in lower plot
    ax2.plot(x2, y1 - y2, color='green', linestyle=' ', marker='.')
    ax2.axhline(0, color='black', linestyle=':')
    ax2.set_xlabel('obs frame wavelength [Angstrom]', fontsize=14)
    ax2.set_ylabel('Observed - model flux \n [ergs/(s cm^2 Angstrom)]', fontsize=14)
```